provider/oci/oci.go

```diff
@@ -236,13 +273,19 @@ func (p *OCIProvider) ApplyChanges(ctx context.Context, changes *plan.Changes) e
 		return nil
 	}
 
-	for zoneID, ops := range opsByZone {
-		if _, err := p.client.PatchZoneRecords(ctx, dns.PatchZoneRecordsRequest{
-			CompartmentId:           &p.cfg.CompartmentID,
-			ZoneNameOrId:            &zoneID,
-			PatchZoneRecordsDetails: dns.PatchZoneRecordsDetails{Items: ops},
-		}); err != nil {
-			return err
+	// Loop zones to get zone scope for zone patch records
+	for _, zone := range zones {
+		for zoneID, ops := range opsByZone {
+			if *zone.Id == zoneID {
+				if _, err := p.client.PatchZoneRecords(ctx, dns.PatchZoneRecordsRequest{
+					CompartmentId:           &p.cfg.CompartmentID,
+					ZoneNameOrId:            &zoneID,
+					PatchZoneRecordsDetails: dns.PatchZoneRecordsDetails{Items: ops},
+					Scope:                   dns.PatchZoneRecordsScopeEnum(zone.Scope),
```

Comment:
You could leave out the scope parameter here too (and compartmentId, although that one was there before).

Reply:
zone scope (private) is required in order to list private zone. I think it have to stay here too.